src/app/parcels/form/clientDatabaseFunctions.ts

```diff
@@ -3,11 +3,14 @@ import { InsertSchema, UpdateSchema } from "@/databaseUtils";
 import { logErrorReturnLogId, logWarningReturnLogId } from "@/logger/logger";
 import { AuditLog, sendAuditLog } from "@/server/auditLog";
 
-type ParcelDatabaseInsertRecord = InsertSchema["parcels"];
-type ParcelDatabaseUpdateRecord = UpdateSchema["parcels"];
+export type ParcelDatabaseInsertRecord = InsertSchema["parcels"];
+export type ParcelDatabaseUpdateRecord = UpdateSchema["parcels"];
```

Comment:
Can we actually instead define and export types for the functions that either insert or update a parcel? Then we can use those in the form props, and also in the function definitions below :)
Lmk if you don't know what I mean

Reply:
e.g put something like
`export type WriteParcelToDatabaseFunction =
    | ((parcelRecord: ParcelDatabaseUpdateRecord) => Promise<UpdateParcelReturnType>)
    | ((parcelRecord: ParcelDatabaseInsertRecord) => Promise<InsertParcelReturnType>);`
    
 in clientDatabaseFunctions.ts

Reply:
Am a little confused how to use it in the function definition below (but can save it in the form type)

Reply:
N.b we still need the ParcelDatabaseUpdateRecord type to be exported as it's needed in Line 140 in EditParcelForm so we can inject the parcelId into the function that we're passing through.

Reply:
I had a look into the line 140 stuff and I can't think of another way to do it so happy to leave it like that for now